File: OnlineStore/OnlineStoreUI/src/model/entities/apiEntities/additional/userRole.ts
export class UserRole{
	role: UserRoleEnum;

	constructor(role:UserRoleEnum){
		this.role = role;
	}

	static getRoleListName(){
        let allStatus:string[]=[
            this.getRolePrint(UserRoleEnum.User),
            this.getRolePrint(UserRoleEnum.Editor),
            this.getRolePrint(UserRoleEnum.Admin)
        ];

        return allStatus;
	}
	
	static getRolePrint(status:UserRoleEnum){

        if(status == UserRoleEnum.User)
            return 'User';
        if(status == UserRoleEnum.Editor)
            return 'Editor';
        if(status == UserRoleEnum.Admin)
            return 'Admin';
               
        return '';
    }
}

export enum UserRoleEnum {
	User = 10,
	Editor = 11,
	Admin = 12
}

File: OnlineStore/Api/CommonEntities/Additional/UserRole.cs
using System;

namespace CommonEntities.Additional
{
	public class UserRole
	{
		public const string RoleUser = "User";
		public const string RoleEditor = "Editor";
		public const string RoleAdmin = "Admin";
		

		public enum RoleEnum
		{
			User = 10,
			Editor = 11,
			Admin = 12
		}

		public RoleEnum Role { get; }

		public UserRole(string role)
		{
			Role = GetRole(role);
		}
		public UserRole(RoleEnum status)
		{
			Role = status;
		}


		public string GetRoleName()
		{
			return GetRoleName(Role);
		}
		
		
		public static string GetRoleName(RoleEnum role)
		{
			switch (role)
			{
				case RoleEnum.User: return RoleUser;
				case RoleEnum.Editor: return RoleEditor;
				case RoleEnum.Admin: return RoleAdmin;
				default: throw new ArgumentOutOfRangeException(nameof(role), role, null);
			}
		}
		public static RoleEnum GetRole(string role)
		{
			switch (role)
			{
				case RoleUser: return RoleEnum.User;
				case RoleEditor: return RoleEnum.Editor;
				case RoleAdmin: return RoleEnum.Admin;
				default: throw new ArgumentOutOfRangeException(nameof(role), role, null);
			}
		}

		

	}
}
